exes/ethercat/inc/public/tfc/ec/config/ethercat.hpp

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <optional>
+#include <string>
+#include <string_view>
+#include <vector>
+
+#include <glaze/core/common.hpp>
+#include <tfc/utils/json_schema.hpp>
+#include <tfc/utils/pragmas.hpp>
+
+#include <tfc/ec/interfaces.hpp>
+
+namespace tfc::ec::config {
+
+struct network_interface {
+  std::string value{};
+
+  struct glaze {
+    static constexpr auto value = &network_interface::value;
+    static constexpr std::string_view name{ "network_interface" };
+  };
+};
+
+}  // namespace tfc::ec::config
+
+template <>
+struct tfc::json::detail::to_json_schema<tfc::ec::config::network_interface> {
+  template <auto Opts>
+  static void op(auto& s, auto&) noexcept {
+    s.oneOf = std::vector<tfc::json::detail::schematic>{};
+
+    // clang-format off
+        PRAGMA_CLANG_WARNING_PUSH_OFF(-Wexit-time-destructors)
+        PRAGMA_CLANG_WARNING_PUSH_OFF(-Wglobal-constructors)
+        thread_local auto interfaces{tfc::global::get_interfaces()};
```

Comment:
maybe it was not elaborated well enough in previous comment. 

In relation to set_interfaces comment, the before mentioned function can be implemented in the source file and store the thread local instance there.
However it does not need to be predeclared in header as a user does not need to call set_interfaces.

What I am trying to say, with reference to `interfaces.cpp`:

```cpp
namespace {
thread_local std::vector<std::string> interfaces{};
}
namespace tfc::global {

auto get_interfaces() -> std::vector<std::string> const& { // or std::span<std::string> or better std::span<std::string_view> if it works
  struct ifaddrs* addrs;
  getifaddrs(&addrs);

  interfaces.clear();

  for (struct ifaddrs* addr = addrs; addr != nullptr; addr = addr->ifa_next) {
    if (addr->ifa_addr && addr->ifa_addr->sa_family == AF_PACKET) {
      interfaces.emplace_back(addr->ifa_name);
    }
  }
  freeifaddrs(addrs);

  return interfaces;
}

}  // namespace tfc::global
```